### PR TITLE
fixes oxyloss not knocking people out at 50

### LINKED ISSUE #79034: Oxyloss no longer causes passing out over 50.
## Testmerges:

The following pull requests are currently test merged:
#78230 : 'Tram v6/Transport Subsystem' by lessthnthree at commit 0117e13ec1

## Reproduction:

1. Put on an air tank.
2. Turn it to 0 kPa and wait.
3. See oxyloss go over 50.
4. Still remain conscious.

I suspect this is due to the changes made to code/modules/mob/living/carbon/carbon_defense.dm in #78657 @vinylspiders 

Particularly these blocks where check_passout() got changed

```
/mob/living/carbon/adjustOxyLoss(amount, updating_health = TRUE, forced, required_biotype, required_respiration_type)
	if(!forced && HAS_TRAIT(src, TRAIT_NOBREATH))
		amount = min(amount, 0) //Prevents oxy damage but not healing

	. = ..()
	check_passout()

/mob/living/carbon/proc/get_interaction_efficiency(zone)
	var/obj/item/bodypart/limb = get_bodypart(zone)
	if(!limb)
		return

/mob/living/carbon/setOxyLoss(amount, updating_health = TRUE, forced, required_biotype, required_respiration_type)
	. = ..()
	check_passout()

/**
* Check to see if we should be passed out from oyxloss
*/
/mob/living/carbon/proc/check_passout()
	if(!isnum(oxyloss))
		return
	if(oxyloss <= 50)
		if(getOxyLoss() > 50)
			ADD_TRAIT(src, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT)
	else if(getOxyLoss() <= 50)
		REMOVE_TRAIT(src, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT)
```

### PR DESCRIPTION
**Pull request by Higgin**
## About The Pull Request

be it far from me to explain why the args not getting put in directly broke this, but reinserting them into the check_passout stuff from #78657 fixes it.

should close #79034 but I have no clue if other similar changes to how args were being handled might not have problems elsewhere.

## Why It's Good For The Game

atmospheric simulator, breathing mechanics good, also chems and i guess bleeding and whatnot

## Changelog

:cl:
fix: automatic breathers rejoice. oxyloss now knocks people out again.
/:cl:

### CODE REVIEW

**Review by vinylspiders — CHANGES_REQUESTED** [review]
This doesn't sound right. Prior to https://github.com/tgstation/tgstation/pull/78657 `.` would be set to `oxyloss` (aka the mob's _oxyloss_ BEFORE applying damage--which was wrong to begin with but it worked, sort of).

Then in https://github.com/tgstation/tgstation/pull/78657 I changed it so `.` is the _net change_ in oxyloss. So with your change you will now only get knocked out if you took 50 or more oxyloss damage in _one go_. I'm not sure how you tested this but yeah..

I think the following changes should fix the issue?

I should really add another unit test for this.

**Comment by Higgin** [comment]
> This doesn't sound right. Prior to #78657 `.` would be set to `oxyloss` (aka the mob's _oxyloss_ BEFORE applying damage--which was wrong to begin with but it worked, sort of).
> 
> Then in #78657 I changed it so `.` is the _net change_ in oxyloss. So with your change you will now only get knocked out if you took 50 or more oxyloss damage in _one go_. I'm not sure how you tested this but yeah..
> 
> I think the following changes should fix the issue?
> 
> I should really add another unit test for this.

I tested it by walking up to a medical kiosk, turning my internals on at 0 kPa, and waiting. It caused me to KO over 50 (I think it hit 52 on that tick.) It worked.

I'll try out these changes and post an update.

**Comment by Higgin** [comment]
Works. Left the !(isnum(mob_oxyloss) check because I'm scared.

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -703,15 +703,14 @@
...
-	if(!isnum(oxyloss))
-		return
-	if(oxyloss <= 50)
-		if(getOxyLoss() > 50)
+	var/mob_oxyloss = getOxyLoss()
+	if(mob_oxyloss >= 50)
+		if(!HAS_TRAIT_FROM(src, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT))
 			ADD_TRAIT(src, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT)
-	else if(getOxyLoss() <= 50)
+	else if(mob_oxyloss < 50)
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/carbon_defense.dm:713]: fuck, this should've been `HAS_TRAIT_FROM(src, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT)`.

Comment (Ghommie) [reply code/modules/mob/living/carbon/carbon_defense.dm:713]: yea

**Pull request merged**